framework/include/userobject/ElementLoopUserObject.h

```diff
@@ -0,0 +1,89 @@
+#ifndef ELEMENTLOOPUSEROBJECT_H
+#define ELEMENTLOOPUSEROBJECT_H
+
+#include "GeneralUserObject.h"
+#include "ThreadedElementLoop.h"
+#include "Coupleable.h"
+#include "MooseVariableInterface.h"
+#include "ScalarCoupleable.h"
+#include "MooseVariableDependencyInterface.h"
+#include "ZeroInterface.h"
+#include "MooseVariable.h"
+#include "SubProblem.h"
+#include "NonlinearSystem.h"
+#include "Assembly.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/parallel_algebra.h"
+
+// Forward Declarations
+class ElementLoopUserObject;
+
+namespace libMesh
+{
+class Elem;
+class QBase;
+}
+
+template<>
+InputParameters validParams<ElementLoopUserObject>();
+
+/**
+ * This will loop over elements and do things
+ *
+ * !!! EXPERIMENTAL !!! Just to see what it takes to do such a thing. Not meant for real use ;-)
+ */
+class ElementLoopUserObject :
+    public GeneralUserObject,
+    public ThreadedElementLoop<ConstElemRange>,
```

Comment:
Hmm, I'm not crazy about this idea but I see why it exists this way. Eventually we'll have to be smarter with all of the looping types. To handle dependency, we might need multiple sweeps through other types. We don't necessarily have to figure all that out before this can go in but I do think the reason for this design should be in the documentation.